%YAML 1.2
# SPDX-FileCopyrightText: NVIDIA CORPORATION & AFFILIATES
# Copyright (c) 2022-2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
# SPDX-License-Identifier: Apache-2.0
---
name: dope_decoder
components:
- name: tensorlist_in
  type: nvidia::gxf::DoubleBufferReceiver
  parameters:
    capacity: 12
- name: detection3darray_out
  type: nvidia::gxf::DoubleBufferTransmitter
  parameters:
    capacity: 12
- name: camera_model_input
  type: nvidia::gxf::DoubleBufferReceiver
  parameters:
    capacity: 12
- name: allocator
  type: nvidia::gxf::UnboundedAllocator
- name: stream_pool
  type: nvidia::gxf::CudaStreamPool
  parameters:
    dev_id: 0
    stream_flags: 1
    stream_priority: 0
    reserved_size: 1
    max_size: 1
    nvtx_identifier: "dope_decoder_stream_pool"
- name: dope_decoder
  type: nvidia::isaac_ros::dope::DopeDecoder
  parameters:
    tensorlist_receiver: tensorlist_in
    camera_model_input: camera_model_input
    detection3darray_transmitter: detection3darray_out
    allocator: allocator
    object_dimensions: []
    object_name: ""
    stream_pool: stream_pool
- type: nvidia::gxf::MessageAvailableSchedulingTerm
  parameters:
    receiver: tensorlist_in
    min_size: 1
- type: nvidia::gxf::MessageAvailableSchedulingTerm
  parameters:
    receiver: camera_model_input
    min_size: 1
- type: nvidia::gxf::DownstreamReceptiveSchedulingTerm
  parameters:
    transmitter: detection3darray_out
    min_size: 1
---
name: sink
components:
- name: signal
  type: nvidia::gxf::DoubleBufferReceiver
  parameters:
    capacity: 2
- type: nvidia::gxf::MessageAvailableSchedulingTerm
  parameters:
    receiver: signal
    min_size: 1
- name: sink
  type: nvidia::isaac_ros::MessageRelay
  parameters:
    source: signal
---
components:
- name: edge0
  type: nvidia::gxf::Connection
  parameters:
    source: dope_decoder/detection3darray_out
    target: sink/signal
---
components:
- name: clock
  type: nvidia::gxf::RealtimeClock
- type: nvidia::gxf::MultiThreadScheduler
  parameters:
    clock: clock
    stop_on_deadlock: false
    check_recession_period_ms: 1
    worker_thread_number: 2
    worker_thread_name_id: "dope_node"
- type: nvidia::gxf::JobStatistics
  parameters:
    clock: clock
